### ISSUE TITLE
Inconsistencies when opening Settings in various window states

### ISSUE THREAD

**Issue by jsejcksn** [atom/tree-view#690]
*(I'm using Atom 1.3.2 on OS X 10.11.2)*

When I attempt to open Settings by pressing <kbd>⌘</kbd>+<kbd>,</kbd> (OS X standard shortcut):

1. in a focused window containing only a new blank document or no document, it opens in a new tab in that window and does not change the toggled state of the tree view or the location of the tree view (toggling tree view has no effect when there is no location set). **:thumbsup: Good behavior**

  ![existing-window](https://cloud.githubusercontent.com/assets/868251/12132064/d190ccbc-b3dd-11e5-8748-7574492565e9.png)

2. in a focused window with a project, it opens in a new tab in that window and does not change the toggled state of the tree view or the location of the tree view. **:thumbsup: Good behavior**

  ![project-window](https://cloud.githubusercontent.com/assets/868251/12132554/54e6984a-b3e2-11e5-9914-e3c20ffd4f4e.png)

3. without a window already open, it opens a new window and toggles the tree view to visible and changes the location of the tree view to `atom://config`. **:thumbsdown: Bad behavior**

  ![new-window](https://cloud.githubusercontent.com/assets/868251/12132084/0015beee-b3de-11e5-9476-b85d702b1f2c.png)

4. with no window focused (but all open windows are minimized), it does nothing. **:thumbsdown: Bad behavior**
  *(but here's a little something because nothing looks pretty boring)*

  ![lol-1](https://cloud.githubusercontent.com/assets/868251/12132579/b77f0302-b3e2-11e5-86a5-b56d337e8990.gif)

---

I noticed that numbers 3 and 4 flashed (briefly highlighted) different menu bar items. 3 flashed `Packages` and 4 flashed `Atom`. This struck me as **very strange** that an app would have different menu items associated with the same shortcut. So I inspected the two menus, finding that (like all other OS X apps), <kbd>⌘</kbd>+<kbd>,</kbd> is assigned to the app's `Preferences` (`Atom > Preferences`), **but also that** it is assigned to `Packages > Settings View > Open`. *(What is this madness?)*

I did more testing.

Here are the results for using `Atom > Preferences` on 1–4:

1a. Same result as 3. :thumbsdown:
2a. Same result as 2. :thumbsup:
3a. Same result as 3.:thumbsdown:
4a. Same result as 3.:thumbsdown:

Here are the results for using `Packages > Settings View > Open` on 1–4:

1b. Same result as 1. :thumbsup:
2b. Same result as 2. :thumbsup:
3b. Not possible (Packages menu is not present) :thumbsdown:
4b. Same result as 4. :thumbsdown:

---

In summary, I think it'd be most appropriate if <kbd>⌘</kbd>+<kbd>,</kbd> were mapped only to `Atom > Preferences` (for continuity with other OS X apps) and that any time it were pressed, it would always open a new tab (if required, in a new window), not modify the toggle state of the tree view, and not change the location of the tree view.

**Comment by jsejcksn** [comment]
I don't know if this is entirely related to tree view or if it is more appropriate for https://github.com/atom/atom/issues. Please advise so I can re-post if necessary.

**Comment by 50Wliu** [comment]
Hey @jsejcksn, I've just created an issue for the first "bad behavior" (point 3) at atom/atom#10297, and I *think* point 4 was just fixed yesterday.  If you're feeling adventurous, feel free to try to build Atom to confirm if it's fixed; otherwise, you'll have to wait a bit until Atom 1.5.0 is released.

/cc'ing @joshaber for point 4 before closing.

**Comment by jsejcksn** [comment]
@50Wliu Do the fixes address the duplication of the shortcut?

**Comment by joshaber** [comment]
Sadly (4) wasn't part of the fixes I did with new windows :disappointed:

**Comment by lee-dohm** [comment]
So is this mostly fixed and maybe we should close this and open a new Issue for just the one missing thing?

**Comment by jsejcksn** [comment]
@lee-dohm No, none of them are fixed yet.

**Comment by jsejcksn** [comment]
Just an update: All behavior I originally described still applies in _Atom 1.5.3 on OS X 10.11.3_

**Comment by Ben3eeE** [comment]
1. Good behavior, nothing to fix.
2. Good behavior, nothing to fix.
3. Fixed by https://github.com/atom/atom/pull/12694.
4. Can't test. Sounds like a mac only issue.

Double keybinding is covered by https://github.com/atom/settings-view/issues/764#issuecomment-206616528 - Feel free to subscribe there for updates with that issue.

Going to close this as resolved, if any of the issues still happen in Atom 1.12 please comment here and we can point you to the correct repository to file it in a new issue.

**Comment by jsejcksn** [comment]
No. 4 is still an issue with 1.10.2 (current release) on macOS 10.12.

I tried to locate a release of 1.12 to download on [the releases page](https://github.com/atom/atom/releases/) for testing, but couldn't—the greatest number I see is 1.11.0-β5. Is anyone else able to test no. 4 on macOS?

/cc @Ben3eeE

**Comment by lee-dohm** [comment]
@jsejcksn I was able to validate item 4 using macOS 10.12. Please open a new Issue for that with detailed repro steps by filling out the Issue template.

**Comment by lee-dohm** [comment]
@jsejcksn Also, please open the new Issue on the Settings View repo at https://github.com/atom/settings-view.

**Comment by jsejcksn** [comment]
@Ben3eeE No. 3 is still an issue in 1.10.2 when getting to settings by using Atom > Preferences.

/cc @lee-dohm (I opened that issue at atom/settings-view#868.)

**Comment by Ben3eeE** [comment]
@jsejcksn Yes, the PR I mentioned is available in Atom 1.12 which currently is in beta.
